### ISSUE TITLE
New integration: Baby Buddy for Android app release

### ISSUE THREAD

**Issue by MrApplejuice** [babybuddy/babybuddy#610]
Hello again cdubz!

Just an heads-up and maybe something for ["the list"](https://github.com/babybuddy/babybuddy#baby-buddy-on-the-web): 

I finally was able to publish the first version of the Android app for BabyBuddy that I was working on for a year now which allows for quicker tracking of timers in BabyBuddy. It is available for free from GitHub via the Releases-page - or for 2 bucks on the Android app store. Links:

- GitHub repository: https://github.com/MrApplejuice/BabyBuddyAndroid/
- Play store link: https://play.google.com/store/apps/details?id=eu.pkgsoftware.babybuddywidgets

Development time was quite slow due to - well -  caring for a newborn over that time interval and me not knowing how many hoops there are to get something published on the Android App Store (not too bad but also not "free"). The app is a bit behind on features, need to work on the new timer-types and tagging :laughing: 

Thanks a lot for the awesome website by the way! We had to track a few things for our newborn in the beginning for some medical reasons and the website was a great help for that! I plan to develop the app a bit further in the near future and continue to contribute to the app's ecosystem/success that way...

And I wanted to at least notify you about the app release myself!

Kind regards,
  Paul K.

------------------

Something else I cannot quite find back: I think you used to have a link to some hosting provider where you could host baby buddy directly. Was that removed? I am not sure if you received a kickback from that or so, but I would be happy to place a link to the hosting provider there if that still exists and makes sense? For the app it would make sense to at least direct someone to an easy-to-find/easy-to-use place to host baby buddy at. Maybe you know a place where people should go for that...?

**Comment by cdubz** [comment]
Exciting!! Congrats!

I haven't tested the play store link yet but I with the direct APK download I am getting an error message, "App not installed as package appears to be invalid".

Will definitely add this to BBoTW list.

> Something else I cannot quite find back: I think you used to have a link to some hosting provider where you could host baby buddy directly. Was that removed? 

Yes it was 😦 It was a Heroku click to install button but I abandoned it after Heroku pulled their support for free tier developer plans. Wasn't getting a kickback or anything and did not want to support that move.

> For the app it would make sense to at least direct someone to an easy-to-find/easy-to-use place to host baby buddy at. Maybe you know a place where people should go for that...?

Yes... that would be nice. And to be fair Heroku _is_ still easy to use. Much easier than other alternatives I looked at a few months back. I think it would be good to dedicate a page in the documentation to easy deployment scenarios like Heroku. Lets consider that work to be done based on this issue.

**Comment by cdubz** [comment]
Oh also meant to mention -- the original logo assets are available under the same license as Baby Buddy itself. Feel free to use them in your Android app if ya want.

https://github.com/babybuddy/babybuddy/tree/master/babybuddy/static_src/logo_assets

**Comment by MrApplejuice** [comment]
:blush: I was about to go to bed, but I could not leave a "broken apk" on the releases page after posting this. Should be fixed now! You can try again, I updated the APK: https://github.com/MrApplejuice/BabyBuddyAndroid/releases/tag/v1.1.5 The signing key was wrong, now you should only get a big old android warning that this is an untrusted app and then selected _Install Anyway_. Hope it works!

> the original logo assets are available under the same license as Baby Buddy itself

Oh wow, thanks a lot. I did not want to copy it verbatim. I think you can see that I at least got some "inspiration" from the original logo. Also the webapp-logo has some problems on Mobile where the icon is downscaled to smaller resolutions and looses too much detail at places (48x48 is the minimum resolution that needs to be supported for the launcher). That is why I simplified the icon a bit. But if you would like it, I can adapt the icon again and make it more similar to the webapp logo, for example set the right colors? Not sure if you would like that?

I felt kind of awkward just copying the "Baby Buddy" name itself and tagging on "for Android". But I came to understand that there are a lot of "baby buddies" out there already and that leaving the naming somewhat similar makes more sense for communication. Again, if you have any objections regarding the naming, please tell me.

**Comment by cdubz** [comment]
> But if you would like it, I can adapt the icon again and make it more similar to the webapp logo, for example set the right colors? Not sure if you would like that?

Entirely up to you! Just wanted to make sure you know you can use the assets 😄 

> I felt kind of awkward just copying the "Baby Buddy" name itself and tagging on "for Android". But I came to understand that there are a lot of "baby buddies" out there already and that leaving the naming somewhat similar makes more sense for communication. Again, if you have any objections regarding the naming, please tell me.

No objections at all!

Although this does make me wonder if I _should_ formalize the app name in some way 🤔 If only because, what if someone else did? Or someone used it for something legitimately malicious?

It's a weird conundrum as I want the whole ecosystem to be as flexible and welcoming as possible. I love to see the cool stuff people build around Baby Buddy!

**Comment by cdubz** [comment]
And confirming the updated APK works 👍  I'll poke around with this soon. Congrats, again!

**Comment by cdubz** [comment]
@MrApplejuice also curious -- are you interested in transferring the Android app to the Baby Buddy GH org and working on it there? No worries if not, understand if you want to keep it under your own account 😄

**Comment by MrApplejuice** [comment]
Hey there! I had to deal with some documentation issues reported by Google Play, so I was a bit caught up with that...

> Although this does make me wonder if I should formalize the app name in some way thinking If only because, what if someone else did? Or someone used it for something legitimately malicious?

I am certainly not an expert on legal issues, especially international legal issues. But yes, I think that might be a good idea for a fairly popular project like this (1.6k stars!!). Maybe something like "media use allowed under [insert license here]" and under the following circumstances [...] could be enough already? Not sure, but even Linux is [registered as a trademark](https://www.linuxfoundation.org/legal/trademark-usage) to avoid issues. I would guess to prevent name poaching or sth... but I am really no expert at this either, sorry.

> And confirming the updated APK works +1 I'll poke around with this soon. Congrats, again!

Thanks! Also again for immediately testing the APK and finding the issue! I am very busy with multiple improvements wrt what events can be tracked and tagging-support, but right now development really has slowed down, because Google Play requires some baby sitting - and of course there were some issues that were revealed after an open release like this...

> are you interested in transferring the Android app to the Baby Buddy GH org and working on it there? No worries if not, understand if you want to keep it under your own account smile

If you are interested to have this app "in the fold", we can do that! Because of my legal name-connection with the app on the Android-store, I need to insist on full control over the repository, however. If that is okay with you, we can migrate the repository, for sure! I think that could be a great idea for the project and increase its visibility.

But maybe you also want to double check if the app actually goes in a direction you like? I am trying to develop some updates, but cannot promise a timeline or anything. I am not sure if there are expectations from your side for what should happen if a project is added to your organization? (quality standards, update, community, etc.)

**Comment by cdubz** [comment]
> I am certainly not an expert on legal issues, especially international legal issues. But yes, I think that might be a good idea for a fairly popular project like this (1.6k stars!!). Maybe something like "media use allowed under [insert license here]" and under the following circumstances [...] could be enough already? Not sure, but even Linux is [registered as a trademark](https://www.linuxfoundation.org/legal/trademark-usage) to avoid issues. I would guess to prevent name poaching or sth... but I am really no expert at this either, sorry.

Yeah, to be clear those were just thinking out loud questions... I wasn't expecting answers 😄 I did do a little research on it and I'm still a bit meh on the topic. I think it makes more sense with a backing _group_ as opposed to individual maintainer. Explicitly stating how this applies to media somewhere may be a good enough change.

> If you are interested to have this app "in the fold", we can do that! Because of my legal name-connection with the app on the Android-store, I need to insist on full control over the repository, however. If that is okay with you, we can migrate the repository, for sure! I think that could be a great idea for the project and increase its visibility.

Great! Lets do it... I think you initiate that on your end? I can't remember what the transfer process looks like exactly. Let me know if you need any guidance.

> But maybe you also want to double check if the app actually goes in a direction you like? I am trying to develop some updates, but cannot promise a timeline or anything. I am not sure if there are expectations from your side for what should happen if a project is added to your organization? (quality standards, update, community, etc.)

This free time, open source development -- I have no expectations! I just think it's great you're working on this now. Whether you are able to in the future... what can ya do? 🤷

**Comment by MrApplejuice** [comment]
Okay... I got a plan for the migration: I want to focus on fixing some bugs with the apps and pushing a major improvement of the app first right now. After I released the set of updates, I will migrate the repository to the babybuddy-org! Will still take a few weeks but I suppose that in May, it is time to get packing and move over :wink:
